### PR TITLE
UpdateTasksByid fixes and Junits Test Cases

### PR DESCRIPTION
**Pull request by LuisTorres07**
Status code when name is invalid fixed,
Bug that brings tasks from different stories fixed,
Junit Test Cases written.

### CODE REVIEW

File: src/test/java/com/stories/service/ServiceTests.java [hunk 1]
```diff
@@ -458,4 +458,61 @@ public void getTaskByIdTry() throws Exception {
 			.thenReturn(testUtils.getDummyTasksDomain());
 		assertEquals(testUtils.getDummyTasksDomain(), storiesServiceImpl.getTaskById(unitTestProperties.getUrlId(), unitTestProperties.getUrlId()));
 	}
+	
+	@Test
+	public void updateTaskbyIdHappyPath() throws Exception{
+		TasksDomain taskDomain = new TasksDomain();
```
Comment (soydianapinto) [inline src/test/java/com/stories/service/ServiceTests.java]: Could you please use a taskDomain from TestUtils?

File: src/test/java/com/stories/service/ServiceTests.java [hunk 2]
```diff
@@ -458,4 +458,61 @@ public void getTaskByIdTry() throws Exception {
...
+		taskDomain.setName("New Tasks Luis");
+		taskDomain.setStatus("Working");
+		when(storiesRepository.existsById(unitTestProperties.getModelId())).thenReturn(storiesApiConstants.getBooleanTrue());
+		when(storiesRepository.findById(unitTestProperties.getModelId())).thenReturn(java.util.Optional.of(testUtils.getStoryTaskModel()));
+		storiesServiceImpl.updateTaskById(taskDomain, unitTestProperties.getModelId(), "5e8cf37b7a605837de2865ad");
```
Comment (soydianapinto) [inline src/test/java/com/stories/service/ServiceTests.java]: Could you please use an id from Constants class?

File: src/test/java/com/stories/service/ServiceTests.java [hunk 3]
```diff
@@ -458,4 +458,61 @@ public void getTaskByIdTry() throws Exception {
...
+	}
+	
+	@Test(expected = EntityNotFoundException.class)
+	public void updateTaskByIdStoryNotFound() throws Exception{
+		TasksDomain taskDomain = new TasksDomain();
```
Comment (VictorMunguiaEnciso) [inline src/test/java/com/stories/service/ServiceTests.java]: Same as diana commen´t you should use a a TaskDomain from the TestUtils.

File: src/test/java/com/stories/service/ServiceTests.java [hunk 4]
```diff
@@ -458,4 +458,61 @@ public void getTaskByIdTry() throws Exception {
...
+		when(storiesRepository.existsById(unitTestProperties.getModelId())).thenReturn(storiesApiConstants.getBooleanFalse());
+		storiesServiceImpl.updateTaskById(taskDomain, unitTestProperties.getModelId(), "5e8cf37b7a605837de2865ad");
+	}
+	
+	@Test(expected = EntityNotFoundException.class)
+	public void updateTaskByIdStatusValidation() throws Exception{
+		TasksDomain taskDomain = new TasksDomain();
```
Comment (VictorMunguiaEnciso) [inline src/test/java/com/stories/service/ServiceTests.java]: Please use a TasksDomain from TestUtils for this and all the Junits below.

**Review by VictorMunguiaEnciso — CHANGES_REQUESTED** [review]

File: src/test/java/com/stories/service/ServiceTests.java (hunk 2 above)
Comment (LuisTorres07) [reply src/test/java/com/stories/service/ServiceTests.java]: fixed

File: src/test/java/com/stories/service/ServiceTests.java (hunk 3 above)
Comment (LuisTorres07) [reply src/test/java/com/stories/service/ServiceTests.java]: fixed

File: src/test/java/com/stories/service/ServiceTests.java (hunk 1 above)
Comment (LuisTorres07) [reply src/test/java/com/stories/service/ServiceTests.java]: fixed

File: src/test/java/com/stories/service/ServiceTests.java [hunk 5]
```diff
@@ -458,4 +458,59 @@ public void getTaskByIdTry() throws Exception {
 			.thenReturn(testUtils.getDummyTasksDomain());
 		assertEquals(testUtils.getDummyTasksDomain(), storiesServiceImpl.getTaskById(unitTestProperties.getUrlId(), unitTestProperties.getUrlId()));
 	}
+	
+	@Test
+	public void updateTaskbyIdHappyPath() throws Exception{
+		TasksDomain taskDomain = testUtils.getDummyTasksDomain();
+		taskDomain.setStatus("Working");
```
Comment (VictorMunguiaEnciso) [inline src/test/java/com/stories/service/ServiceTests.java]: Move this "working" value to constants class please

File: src/test/java/com/stories/service/ServiceTests.java [hunk 6]
```diff
@@ -458,4 +458,59 @@ public void getTaskByIdTry() throws Exception {
...
+	public void updateTaskByIdStatusValidation() throws Exception{
+		TasksDomain taskDomain = testUtils.getDummyTasksDomain();
+		when(storiesRepository.existsById(unitTestProperties.getModelId())).thenReturn(storiesApiConstants.getBooleanTrue());
+		storiesServiceImpl.updateTaskById(taskDomain, unitTestProperties.getModelId(), storiesApiConstants.getValidPutTaskId());
+	}
+	
+	@Test(expected = EntityNotFoundException.class)
+	public void updateTaskbyIdInvalidAsignee() throws Exception{
+		TasksDomain taskDomain = testUtils.getDummyTasksDomain();
+		taskDomain.setStatus("Working");
```
Comment (VictorMunguiaEnciso) [inline src/test/java/com/stories/service/ServiceTests.java]: Same with this value

File: src/test/java/com/stories/service/ServiceTests.java [hunk 7]
```diff
@@ -458,4 +458,59 @@ public void getTaskByIdTry() throws Exception {
...
+		taskDomain.setAssignee("5e6bbc854244ac0cbc8df65d");
```
Comment (VictorMunguiaEnciso) [inline src/test/java/com/stories/service/ServiceTests.java]: And this one too please

File: src/test/java/com/stories/service/ServiceTests.java [hunk 8]
```diff
@@ -458,4 +458,59 @@ public void getTaskByIdTry() throws Exception {
...
+		when(storiesRepository.existsById(unitTestProperties.getModelId())).thenReturn(storiesApiConstants.getBooleanTrue());
+		when(usersRepository.existsById(taskDomain.getAssignee())).thenReturn(storiesApiConstants.getBooleanFalse());
+		storiesServiceImpl.updateTaskById(taskDomain, unitTestProperties.getModelId(), storiesApiConstants.getValidPutTaskId());
+	}
+	
+	@Test(expected = EntityNotFoundException.class)
+	public void updateTaskbyIdNameEmpty() throws Exception{
+		TasksDomain taskDomain = testUtils.getDummyTasksDomain();
+		taskDomain.setName(null);
+		taskDomain.setStatus("Working");
```
Comment (VictorMunguiaEnciso) [inline src/test/java/com/stories/service/ServiceTests.java]: this "working" too

File: src/test/java/com/stories/service/ServiceTests.java [hunk 9]
```diff
@@ -458,4 +458,59 @@ public void getTaskByIdTry() throws Exception {
...
+		taskDomain.setAssignee("5e6bbc854244ac0cbc8df65d");
```
Comment (VictorMunguiaEnciso) [inline src/test/java/com/stories/service/ServiceTests.java]: This value too

File: src/test/java/com/stories/service/ServiceTests.java [hunk 10]
```diff
@@ -458,4 +458,59 @@ public void getTaskByIdTry() throws Exception {
...
+		when(storiesRepository.existsById(unitTestProperties.getModelId())).thenReturn(storiesApiConstants.getBooleanTrue());
+		when(usersRepository.existsById(taskDomain.getAssignee())).thenReturn(storiesApiConstants.getBooleanTrue());
+		storiesServiceImpl.updateTaskById(taskDomain, unitTestProperties.getModelId(), storiesApiConstants.getValidPutTaskId());
+	}
+	
+	@Test(expected = EntityNotFoundException.class)
+	public void updateTaskbyIdInvalidTask() throws Exception{
+		TasksDomain taskDomain = testUtils.getEmptyTasksDomain();
+		taskDomain.setName("Test");
```
Comment (VictorMunguiaEnciso) [inline src/test/java/com/stories/service/ServiceTests.java]: Same here with "Test" and "Working"

**Review by VictorMunguiaEnciso — CHANGES_REQUESTED** [review]

**Review by VictorMunguiaEnciso — APPROVED** [review]

**Review by VictorMunguiaEnciso — CHANGES_REQUESTED** [review]
you have some conflicts

**Review by VictorMunguiaEnciso — DISMISSED** [review]

**Review by VictorMunguiaEnciso — APPROVED** [review]

**Pull request merged**